src/lib.rs

```diff
@@ -76,32 +196,77 @@ impl TempDir {
                        "too many temporary directories already exist"))
     }
 
-    /// Attempts to make a temporary directory inside of `env::temp_dir()` whose
-    /// name will have the prefix `prefix`. The directory will be automatically
-    /// deleted once the returned wrapper is destroyed.
+    /// Accesses the [`Path`] to the temporary directory.
     ///
-    /// If no directory can be created, `Err` is returned.
-    pub fn new(prefix: &str) -> io::Result<TempDir> {
-        TempDir::new_in(&env::temp_dir(), prefix)
+    /// [`Path`]: http://doc.rust-lang.org/std/path/struct.Path.html
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use tempdir::TempDir;
+    ///
+    /// let tmp_dir = TempDir::new("example").expect("create temp dir");
+    ///
+    /// // Check that the temp directory actually exists.
+    /// assert!(tmp_dir.path().exists());
+    /// ```
+    pub fn path(&self) -> &path::Path {
+        self.path.as_ref().unwrap()
     }
 
-    /// Unwrap the wrapped `std::path::Path` from the `TempDir` wrapper.
-    /// This discards the wrapper so that the automatic deletion of the
-    /// temporary directory is prevented.
+    /// Unwraps the [`Path`] contained in the `TempDir` and
+    /// returns it. This destroys the `TempDir` without deleting the
+    /// directory represented by the returned `Path`.
+    ///
+    /// [`Path`]: http://doc.rust-lang.org/std/path/struct.Path.html
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::fs;
+    /// use tempdir::TempDir;
+    ///
+    /// let tmp_dir = TempDir::new("example").expect("create temp dir");
+    ///
+    /// // Convert `tmp_dir` into a `Path`, destroying the `TempDir`
+    /// // without deleting the directory.
+    /// let tmp_path = tmp_dir.into_path();
+    ///
+    /// // Delete the temporary directory ourselves.
+    /// fs::remove_dir_all(tmp_path).expect("remove temp dir");
+    /// ```
     pub fn into_path(mut self) -> PathBuf {
         self.path.take().unwrap()
     }
 
-    /// Access the wrapped `std::path::Path` to the temporary directory.
-    pub fn path(&self) -> &path::Path {
-        self.path.as_ref().unwrap()
-    }
-
-    /// Close and remove the temporary directory
+    /// Closes and removes the temporary directory, returing a `Result`.
     ///
     /// Although `TempDir` removes the directory on drop, in the destructor
     /// any errors are ignored. To detect errors cleaning up the temporary
     /// directory, call `close` instead.
+    ///
```

Comment:
Here, an "Errors" section might be especially helpful even if just for mentioning usual errors and linking to `io::Result`.